packages/lisk-p2p/src/peer.ts

```diff
@@ -572,3 +572,149 @@ export class Peer extends EventEmitter {
 		request.end(legacyNodeInfo);
 	}
 }
+
+export interface ConnectAndFetchResponse {
+	readonly responsePacket: P2PResponsePacket;
+	readonly socket?: SCClientSocket;
+}
+
+export const connectAndRequest = async (
+	basicPeerInfo: P2PPeerInfo,
+	procedure: string,
+	nodeInfo?: P2PNodeInfo,
+	peerConfig?: PeerConfig,
+): Promise<ConnectAndFetchResponse> =>
+	new Promise<ConnectAndFetchResponse>(
+		(
+			resolve: (result: ConnectAndFetchResponse) => void,
+			reject: (result: Error) => void,
+		): void => {
+			const legacyNodeInfo = nodeInfo
+				? convertNodeInfoToLegacyFormat(nodeInfo)
+				: undefined;
+			// Add a new field discovery to tell the receiving side that the connection will be short lived
+			const nodeInfoForDiscovery = { discovery: true, ...legacyNodeInfo };
+			const requestPacket = {
+				procedure,
+			};
+			const clientOptions: ClientOptionsUpdated = {
+				hostname: basicPeerInfo.ipAddress,
+				port: basicPeerInfo.wsPort,
+				query: querystring.stringify(nodeInfoForDiscovery),
+				connectTimeout: peerConfig
+					? peerConfig.connectTimeout
+						? peerConfig.connectTimeout
+						: DEFAULT_CONNECT_TIMEOUT
+					: DEFAULT_CONNECT_TIMEOUT,
+				ackTimeout: peerConfig
+					? peerConfig.connectTimeout
+						? peerConfig.connectTimeout
+						: DEFAULT_CONNECT_TIMEOUT
+					: DEFAULT_ACK_TIMEOUT,
+				multiplex: false,
+				autoConnect: false,
+				autoReconnect: false,
+				pingTimeoutDisabled: true,
+			};
+
+			const outboundSocket = socketClusterClient.create(clientOptions);
+			// Attaching handlers for various events that could be used future for logging or any other application
+			outboundSocket.on('error', async (error: Error) =>
+				Promise.resolve(error),
+			);
+			outboundSocket.on('close', async () =>
+				Promise.resolve('Connection closed'),
```

Comment:
same as above, also it should rejected with `Error` object?

Reply:
will be handled in Peer object now